spec/models/user_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe User do
+  let(:valid_user_params) { { email: "bob@example.org", password: "secret" } }
+
+  it "is valid with valid params" do
+    expect(User.new(valid_user_params)).to be_valid
+  end
+
+  it "is invalid without an email" do
+    expect(User.new(valid_user_params.merge!(email: nil))).not_to be_valid
+  end
+
+  it "is invalid with a blank email" do
+    expect(User.new(valid_user_params.merge!(email: ""))).not_to be_valid
+  end
+
+  it "is invalid with an incorrect email" do
+    expect(User.new(valid_user_params.merge!(email: "bob-example"))).not_to be_valid
+  end
+
+  it "is invalid with a non-unique email" do
+    User.create(valid_user_params)
+    expect(User.new(valid_user_params.merge!(email: valid_user_params[:email]))).not_to be_valid
+  end
+
+  it "is invalid without a password" do
+    expect(User.new(valid_user_params.merge!(password: nil))).not_to be_valid
+  end
+
+  it "is invalid with a blank password" do
+    expect(User.new(valid_user_params.merge!(password: ""))).not_to be_valid
+  end
+
+  it "is invalid with a password of less then 6 chars" do
+    expect(User.new(valid_user_params.merge!(password: "12345"))).not_to be_valid
+  end
+
+  it "is invalid with a password of more then 128 chars" do
```

Comment:
Yeah, I know, never knew how to go about it. Your way has better readability when using the `-f d` rspec output, but then the pseudo-english in `_spec.rb` itself it makes no sense.